src/plugins/query_enhancements/server/routes/data_source_connection/routes.ts

```diff
@@ -31,7 +31,12 @@ export function registerDataSourceConnectionsRoutes(
         if (error.statusCode === 404 || error.statusCode === 400) {
           return response.ok({ body: [] });
         }
-        return response.custom({ statusCode: error.statusCode || 500, body: error.message });
+        // Transform 500 errors to 503 to indicate service availability issues
+        const statusCode = error.statusCode === 500 ? 503 : error.statusCode || 503;
```

Comment:
Are all 500s 503s?

Reply:
Currently we only see 500 "No Living connections". Might have other 5** errors.

Reply:
OSD should only return 500 if itself runs into internal server error. Here if an `error` has a `.statusCode` that is 500, the error must be coming from a different service that had internal server error. They are actual 500s in other services, not Neo